deluge/ui/ui_entry.py

```diff
@@ -35,11 +49,35 @@ def start_ui():
 
     # Get the registered UI entry points
     ui_entrypoints = {}
-    for entrypoint in pkg_resources.iter_entry_points('deluge.ui'):
+    if _use_pkg_resources:
+        # Legacy method using pkg_resources
+        for entrypoint in pkg_resources.iter_entry_points('deluge.ui'):
+            try:
+                ui_entrypoints[entrypoint.name] = entrypoint.load()
+            except ImportError:
+                # Unable to load entrypoint so skip adding it.
+                pass
+    else:
+        # Modern method using importlib.metadata
         try:
-            ui_entrypoints[entrypoint.name] = entrypoint.load()
-        except ImportError:
-            # Unable to load entrypoint so skip adding it.
+            eps = entry_points(group='deluge.ui')
```

Comment:
The `group` attribute is passed down to `select` so later call is redundant. Also `group` is a feature of Python 3.10 so this does break on Python 3.9. 

We can simplify quite a bit by EAFP, I think the following should work

```
try:
   entrypoints = entry_points(group='deluge.ui')
except TypeError:
   try:
      entrypoints = list(set(entry_points()['deluge.ui']))
   except KeyError:
      entrypoints = []

for entrypoint in entrypoints:
...
```

I have update the CI test to use 3.9 and 3.13 so should catch any problems